base_layer/core/src/mempool/mempool_storage.rs

```diff
@@ -220,17 +233,18 @@ impl MempoolStorage {
     }
 
     /// Gathers and returns the stats of the Mempool.
-    pub fn stats(&self) -> Result<StatsResponse, MempoolError> {
+    pub fn stats(&mut self) -> Result<StatsResponse, MempoolError> {
```

Comment:
Because of the way TtlCache works, specifically, the way it clears out expired items when you call some normally read-only methods like `len()` and `get()`. Before we had a double lock on the mempool storage which doesn't do much. The only downside is a need for a write lock on the mempool instead of just a read when calling stats and state - but these calls aren't performance critical (i.e. pretty much only called from the bn prompt)